ClientTests/MenuTests.swift

```diff
@@ -24,73 +24,286 @@ class MenuTests: XCTestCase {
-        XCTAssertEqual(browserConfiguration.menuItems[4].title, AppMenuConfiguration.NewPrivateTabTitleString)
-        XCTAssertEqual(browserConfiguration.menuItems[5].title, AppMenuConfiguration.AddBookmarkTitleString)
-        XCTAssertEqual(browserConfiguration.menuItems[6].title, AppMenuConfiguration.SettingsTitleString)
-
-        XCTAssertNotNil(browserConfiguration.menuToolbarItems)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems!.count, 4)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems![0].title, AppMenuConfiguration.TopSitesTitleString)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems![1].title, AppMenuConfiguration.BookmarksTitleString)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems![2].title, AppMenuConfiguration.HistoryTitleString)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems![3].title, AppMenuConfiguration.ReadingListTitleString)
-
-
-        tabState = TabState(isPrivate: true, desktopSite: true, isBookmarked: true, url: NSURL(string: "http://mozilla.com")!, title: "Mozilla", favicon: nil)
-        browserConfiguration = AppMenuConfiguration(appState: appState(.Tab(tabState: tabState)))
-        XCTAssertEqual(browserConfiguration.menuItems.count, 7)
-        XCTAssertEqual(browserConfiguration.menuItems[0].title, AppMenuConfiguration.FindInPageTitleString)
-        XCTAssertEqual(browserConfiguration.menuItems[1].title, AppMenuConfiguration.ViewMobileSiteTitleString)
-        XCTAssertEqual(browserConfiguration.menuItems[2].title, AppMenuConfiguration.OpenHomePageTitleString)
-        XCTAssertEqual(browserConfiguration.menuItems[3].title, AppMenuConfiguration.NewTabTitleString)
-        XCTAssertEqual(browserConfiguration.menuItems[4].title, AppMenuConfiguration.NewPrivateTabTitleString)
-        XCTAssertEqual(browserConfiguration.menuItems[5].title, AppMenuConfiguration.RemoveBookmarkTitleString)
-        XCTAssertEqual(browserConfiguration.menuItems[6].title, AppMenuConfiguration.SettingsTitleString)
-
-
-        XCTAssertNotNil(browserConfiguration.menuToolbarItems)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems!.count, 4)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems![0].title, AppMenuConfiguration.TopSitesTitleString)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems![1].title, AppMenuConfiguration.BookmarksTitleString)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems![2].title, AppMenuConfiguration.HistoryTitleString)
-        XCTAssertEqual(browserConfiguration.menuToolbarItems![3].title, AppMenuConfiguration.ReadingListTitleString)
-    }
-
-    func testMenuConfigurationForHomePanels() {
+    // bookmarks menu item
+    func testMenuConfigurationForNonBookmarkedItemInTab() {
+        let tabState = TabState(isPrivate: false, desktopSite: false, isBookmarked: false, url: NSURL(string: "http://mozilla.com")!, title: "Mozilla", favicon: nil)
```

Comment:
No content is loaded from the URL, so it doesn't really matter, as long as its a valid URL.